FreeSurfer/FreeSurferPipeline.sh

```diff
@@ -204,12 +204,18 @@ PARAMETERs are: [ ] = optional; < > = user supplied value
             (ii) you want to be able to run some flag in recon-all, without also regenerating the surfaces.
                  e.g., [--existing-subject --extra-reconall-arg=-show-edits --no-conf2hires]
 
+  [--mpp-mode={HCPStyleData, LegacyStyleData}]
+      Indicates, which variant of MPP to use. "HCPStyleData" (the default) follows the processing steps 
```

Comment:
@coalsont Do you think we should spell out what "MPP" is as part of the usage statement?
i.e., "Indicates which variant of MPP (minimal preprocessing) to use."

Reply:
I seem to recall there was some discussion about retconning it into "multimodal processing pipelines" or something.

I have mixed feelings about that acronym (we have yet to use it within the pipelines, this option would be the first, but we have used the acronym MMP before for something entirely different), and if I were writing only the usage statement, I'd probably sidestep the issue by saying something like `Indicate whether the HCP acquisition and processing guidelines should be treated as requirements`.

Reply:
Ok. I'll ask you for a consensus here, as this would need to be changed in multiple PRs and I would not like to go back and forth on it. @mharms @coalsont @glasserm please provide a final guideline.

Reply:
I like that, but it of course begs the question of why the flag is called `--mpp-mode`, which is why I broached the question earlier (via email) if `--pipeline-mode` wouldn't in fact be a (much) better name... :)  I feel that if the flag is going to be called `--mpp-mode` then we really need to explain what "MPP" actually stands for.

Reply:
I have added that explanation to PreFS:

```
  [--mpp-mode=(HCPStyleData|          Which variant of Minimal Preprocessing Pipelines (MPP) to use. "HCPStyleData" 
               LegacyStyleData)]      (the default) follows the processing stepsdescribed in Glasser et al. (2013) 
                                      and requires 'HCP-Style' data acquistion. "LegacyStyleData" allows additional 
```

Reply:
I think --pipeline-mode is better than --mpp-mode, as the tool name doesn't seem like something that belongs in an option.  Another option might be --processing-mode as more specific and less redundant, as "pipeline" is still a name for the overall thing that is being used, and is therefore somewhat redundant (e.g., I didn't make a `wb_command -workbench-version` option, it is just `wb_command -version`).

Reply:
I like `--processing-mode=(HCPStyleData|LegacyStyleData)`.  @glasserm @aanticev Is that acceptable to you?  If so, I think we finally have consensus.  [We either need to explain what "MPP" refers to in Usage statements, or simply drop "mpp" from the argument name, which is the approach that both Tim and I favor].

Reply:
In case he doesn't explicitly send an email or make a comment, @glasserm told me in person he is fine with `--processing-mode` (he did not specifically mention the values it accepts, but I think he is okay with the ones you mentioned).

Reply:
[--processing-mode=(HCPStyleData|LegacyStyleData)] works for me.

Reply:
So, if we go that route, presumably we call the new shlib `processingmodecheck.shlib` (rather than `mppmodecheck.shlib`), and the function therein (and called by the scripts) `check_mode_compliance()` (or alternatively `processing_mode_compliance()`), and the internal variable used within the scripts `ProcessingMode` (rather than `MPPMode`)?  That would seem to tie everything together nicely and consistently.  Any objections to those (or better suggestions)? @coalsont @glasserm

Reply:
Sounds good.

Reply:
Sounds reasonable, though I prefer that the function name still starts with "check".

Reply:
So:
-> processingmodecheck.shlib 
-> check_mode_compliance() 
-> ProcessingMode (rather than MPPMode)

I'll implement that throughout

Reply:
Also remove "MPP" from the help info.  You can start from my earlier suggestion of `Indicate whether the HCP acquisition and processing guidelines should be treated as requirements` and rephrase/expand to include the default and accepted values.

Reply:
I took out MPP from help and comments. I did not know we are changing the wording @glasserm provided and @mharms polished. If we are changing them, can I please get the final agreed upon wording so that there's no back and forth.

Reply:
How about:
```
Controls whether the HCP acquisition and processing guidelines should be treated as requirements.
"HCPStyleData" (the default) follows the processing steps described in Glasser et al. (2013) 
   and requires 'HCP-Style' data acquistion. 
"LegacyStyleData" allows additional processing functionality and use of some acquisitions
   that do not conform to 'HCP-Style' expectations.
   i.e., in this case no high-resolution T2w image acquisition
```
Note that I intentionally lead with the mode names as the start of new lines, and used indentation purposely to improve readability.

Reply:
Ok. Changed it here and in PreFS.

Reply:
Looks reasonable, though I still don't like the seeming redundancy in "i.e., in this case" or that it is a sentence fragment after a period.  How about "In this script, it allows not having a high-resolution T2w image."?

Reply:
Changed.